worlds/oot_soh/location_access/overworld/market.py

```diff
@@ -5,8 +5,272 @@
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT9, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT10, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT11, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE1, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE2, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE3, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE4, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE5, lambda bundle: (is_child(bundle) and can_break_crates(bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_GUARD_HOUSE, world, [
+        (Regions.MARKET_ENTRANCE, lambda bundle: True)
+    ])
+    
+    ## Market Bazaar
+    # Locations
+    add_locations(Regions.MARKET_BAZAAR, world, [
+        (Locations.MARKET_BAZAAR_ITEM1, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM2, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM3, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM4, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM5, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM6, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM7, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM8, lambda bundle: True),
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_BAZAAR, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Mask Shop
+    # Events
+    add_events(Regions.MARKET_MASK_SHOP, world, [
+        # Todo: Is this accurate? CAN_BORROW_MASKS,BORROW_SKULL_MASK in Events, rest missing. Add to Enums?
+        (EventLocations.MARKET_MASK_SHOP, Events.CAN_BORROW_MASKS, lambda bundle: (has_item(Items.ZELDAS_LETTER, bundle) and world.options.kakariko_gate.value))
+        (EventLocations.MARKET_MASK_SHOP, Events.BORROW_SKULL_MASK, lambda bundle: (world.options.complete_mask_quest.value and Events.CAN_BORROW_MASKS)),
+        (EventLocations.MARKET_MASK_SHOP, Events.BORROW_SPOOKY_MASK, lambda bundle: (world.options.complete_mask_quest.value and Events.CAN_BORROW_MASKS)),
+        (EventLocations.MARKET_MASK_SHOP, Events.BORROW_BUNNY_HOOD, lambda bundle: (world.options.complete_mask_quest.value and Events.CAN_BORROW_MASKS)),
+        (EventLocations.MARKET_MASK_SHOP, Events.BORROW_RIGHT_MASKS, lambda bundle: (world.options.complete_mask_quest.value and Events.CAN_BORROW_MASKS)),
+    ])
+    # Locations
+    add_locations(Regions.MARKET_MASK_SHOP, world, [
+        (Locations.MARKET_MASK_SHOP_HINT, lambda bundle: True)
+    ])
```

Comment:
This location is only used for holding a hint in SoH, you can remove this one entirely.